stan/math/prim/scal/fun/value_of_rec.hpp

```diff
@@ -1,6 +1,8 @@
 #ifndef STAN_MATH_PRIM_SCAL_FUN_VALUE_OF_REC_HPP
 #define STAN_MATH_PRIM_SCAL_FUN_VALUE_OF_REC_HPP
 
+#include <Eigen/Dense>
```

Comment:
@charlesm93: things inside `prim` cannot bring in any of Eigen's headers. This header include needs to be removed from this file.